.azure-pipelines/client.yml

```diff
@@ -0,0 +1,83 @@
+trigger:
+  - master
+
+variables:
+  node.version: '10.x'
+  build.path: 'lib/services/keyVault'
+
+jobs:
+  - job: 'Publish'
+
+    pool:
+      vmImage: 'ubuntu-16.04'
+
+    steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: '$(node.version)'
+        displayName: 'Install Node.js $(node.version)'
+
+      - task: Npm@1
+        inputs:
+          command: 'install'
+          workingDir: '$(build.path)'
+        displayName: 'npm install'
+
+      - task: Npm@1
+        inputs:
+          command: 'custom'
+          customCommand: 'pack'
+          workingDir: '$(build.path)'
+        displayName: 'npm pack'
+
+      - task: CopyFiles@2
+        inputs:
+          sourceFolder: '$(build.path)'
+          contents: '*.tgz'
+          targetFolder: $(Build.ArtifactStagingDirectory)
+        displayName: 'Copy Package'
+
+      - task: PublishBuildArtifacts@1
+        displayName: 'Publish Package'
+
+  - job: 'Audit'
+
+    pool:
+      vmImage: 'ubuntu-16.04'
+
+    steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: '$(node.version)'
+        displayName: 'Install Node.js $(node.version)'
+
+      - task: Npm@1
+        inputs:
+          command: 'custom'
+          customCommand: 'i --package-lock-only'
+          workingDir: '$(build.path)'
+        displayName: 'npm i --package-lock-only'
+
+      - powershell: "npm audit --json | ConvertFrom-Json | Select-Object  -expandProperty 'metadata' | select-object -expandproperty 'vulnerabilities' | %{ if ($_.low + $_.moderate + $_.high + $_.critical -ne 0) { write-error 'FAILED! Your package has vulnerabilities. Check npm audit' }}"
```

Comment:
Does Ubuntu image on DevOps come with PowerShell preinstalled? 

Also, nice check. We should port this to every JS project instead of looking for vulnerabilities manually.

Reply:
Both Ubuntu and Mac images include PowerShell 6.1 by default:

https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/linux/Ubuntu1604-README.md
https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/macos/macos-Readme.md

We're planning to use PowerShell as our default xplat language for most scripts.